src/incubator/TextField/index.tsx

```diff
@@ -124,18 +126,42 @@ const TextField = (props: InternalTextFieldProps) => {
     return [inputStyle, styles.dummyPlaceholder];
   }, [inputStyle]);
 
-  return (
-    <FieldContext.Provider value={context}>
-      <View {...containerProps} style={[margins, positionStyle, containerStyle, centeredContainerStyle]}>
+  let textFieldLabel = (
+    <Label
```

Comment:
Instead of handling this logic in the main index, move it inside the Label component, so it's more encapsulated and less messy for the index file. 

Moreover, consider not rendering two labels, but concatenate the * with the original label
I would not color the * with red, today we already have a solution for this in private (in TextField.config) where we don't color the * so better keep it aligned with our UI guidelines. 

If you still want to support coloring the asterisk for the community (was it even requested by the user?) there are a bit simpler ways doing that..

Reply:
Done. I had to pass the showMandatoryIndication to the label though.
On regard to the red asterisk there was an image attached to the issue where the asterisk was red so I thought this is what we wanted. I removed the coloring for now.